### PR TITLE
Page list template

### PR DESCRIPTION
**Pull request by hectoruch**
- Create URL page list **(/pages)**.
- Create template.
- Add styles.
- Add functions to API and get data.
- Pages end point API.
- Pages categories end point API.

image

### CODE REVIEW

**Comment by hectoruch** [comment]
- Fix loader styles.
- Add pagination.

image

File: sites/all/themes/custom/ogp_theme/sass/main.scss [hunk 1]
```diff
@@ -28,6 +28,7 @@
 'layouts/l-news-events',
 'layouts/l-working-group',
 'layouts/l-working-group-list',
+'layouts/l-pages-list',
```
Comment (edbrett) [inline sites/all/themes/custom/ogp_theme/sass/main.scss]: If you already have the same view for groups, why do you need another css view?? Please reuse code.

File: sites/all/themes/custom/ogp_theme/lib/layouts/pages/pagesList.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        } else {
+          createTable(pageresult);
+          removeLoader('#tableContainer', null, true);
+          initPagination(pageNumber, totalPages, 'pagesList');
+          setPaginationListerners();
+        }
+      });
+    }
+
+    function createTable(data) {
```
Comment (edbrett) [inline sites/all/themes/custom/ogp_theme/lib/layouts/pages/pagesList.js]: Can this not be a general function for both groups and pages? And any other lists/table we created. Less code is better.

File: sites/all/themes/custom/ogp_theme/lib/layouts/pages/pagesList.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      let html = '';
+      for (let i = 0; i < data.data.length; i += 1) {
+        if (data.data[i].body) {
+          html += `
+            <tr class="text">
+              <th class="text -small-bold">${data.data[i].label}</th>
+              <th class="text">${data.data[i].page_category.label}</th>
+              <th class="text -blue -capitalize">${moment(parseInt(data.data[i].date) * 1000).startOf('month').fromNow()}</th>
```
Comment (edbrett) [inline sites/all/themes/custom/ogp_theme/lib/layouts/pages/pagesList.js]: You can use moment.js to get this very easily. You have used elsewhere. Lets choose one or the other.

**Comment by hectoruch** [comment]
- Add table.js general function.
- Improve moment.js implementation.
- Update styles files.

**Pull request merged**